lib/rspec/mocks/proxy.rb

```diff
@@ -161,7 +161,19 @@ def has_negative_expectation?(message)
       # @private
       def record_message_received(message, *args, &block)
         @order_group.invoked SpecificMessage.new(object, message, args)
-        @messages_received << [message, args.map(&:dup), block]
+        @messages_received << [message, args.map { |arg| dup_if_mutable(arg) }, block]
+      end
+
+      # @private
+      def dup_if_mutable(arg)
```

Comment:
Do you think it's worth refactoring this method to make it clear that we only `dup`, `Enumerable` and `String`. And everything else we don't `dup`. 

In essence what I'm saying is remove the first `when IO, StringIO` and just refactor this code to say something like:

```
case arg
when Enumerable, String
  arg.dup 
else
  arg
end
```

Reply:
Both `IO` and `StringIO` are _also_ `Enumerable` :(

Reply:
Thanks @xaviershay, I think that should be included in the commit message. 

Code comments normally expire, whereas commit messages are like living documentation for why some code was written in a certain way. Perhaps we could replace the comment with a commit message?

Reply:
eh I don't have strong feelings, but bikeshedding fwiw.. :)

For me, commit message = explain why we made changes in the first place. Comments = explain why/how a particular bit of (small) code works. I don't see how this comment would "expire" since it's hyperlocal to the code it affects. This is also a pattern we need to use multiple times in the code base [citation needed, but I know I've been bitten by it a few times].

Reply:
It's clear enough and matches how we do it elsewhere